In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/validation.csv
/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/train.csv
/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/test.csv


# TEXT SUMMARIZATION

In [3]:
import csv

def load_doc_from_csv(filename, text_column):
    """
    texts = load_doc_from_csv('/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/train.csv', 1)

    Args:
    - filename (str): The path to the CSV file.
    - text_column (int): The index of the column containing the text in the CSV file.

    Returns:
    - List[str]: A list of texts loaded from the CSV file.
    """
    texts = []
    with open(filename, 'r', encoding='utf-8') as file:
        csv_reader = csv.reader(file)
        for row in csv_reader:
            if len(row) > text_column:
                texts.append(row[text_column])
    return texts

In [4]:
import os

def load_stories(directory):
    """
    texts = load_doc_from_csv('/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/train.csv', 1)
    
    Args:
    - directory (str): The directory containing the text files.
    
    Returns:
    - List[str]: A list of documents loaded from the files.
    """
    documents = []
    for name in os.listdir(directory):
        filename = os.path.join(directory, name)
        if os.path.isfile(filename):
            # load document
            with open(filename, 'r', encoding='utf-8') as file:
                document = file.read()
                documents.append(document)
    return documents

In [5]:
import os

def load_stories(directory):
    """
    texts = load_doc_from_csv('/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/train.csv', 1)

    Args:
    - directory (str): The directory containing the text files.

    Returns:
    - List[str]: A list of documents loaded from the files.
    """
    documents = []
    for name in os.listdir(directory):
        filename = os.path.join(directory, name)
        # Check if the path is a file
        if os.path.isfile(filename):
            # Load document
            with open(filename, 'r', encoding='utf-8') as file:
                doc = file.read()
                documents.append(doc)
    return documents

In [6]:
def split_story(doc):
    """
    Split a document into news story and highlights.

    Args:
    - doc (str): The document containing the story and highlights.

    Returns:
    - Tuple[str, List[str]]: A tuple containing the story and a list of highlights.
    """
    # Find the first occurrence of '@highlight'
    index = doc.find('@highlight')
    
    # Split into story and highlights
    story, highlights = doc[:index], doc[index:].split('@highlight')
    
    # Strip extra white space around each highlight
    highlights = [h.strip() for h in highlights if len(h.strip()) > 0]
    
    return story.strip(), highlights

> Load from Article

In [20]:
import os
import pandas as pd

def load_doc(filename):
    """
    Load text from a file.

    Args:
    - filename (str): The path to the file.

    Returns:
    - str: Text content of the file.
    """
    with open(filename, 'r', encoding='utf-8') as file:
        text = file.read()
    return text

def load_stories(filename):
    """
    Load a story from a file.

    Args:
    - filename (str): The path to the file containing the story.

    Returns:
    - dict: A dictionary containing the story and highlights.
    """
    # Load CSV file
    df = pd.read_csv(filename)
    stories = []
    for article in df['article']:
        # Split into story and highlights
        story, highlights = split_story(article)
        stories.append({'story': story, 'highlights': highlights})
    return stories

# Load stories
file_path = '/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/train.csv'
stories = load_stories(file_path)
print('Loaded Article Stories:', len(stories))

Loaded Article Stories: 287113


* The number 287,113 represents the count of such articles that have been successfully processed and loaded into memory or a data structure.

In [21]:
import os

def load_stories(file_path):
    """
    Load stories from a file.

    Args:
    - file_path (str): The path to the file containing the stories.

    Returns:
    - List[dict]: A list of dictionaries containing story and highlights.
    """
    # Check if the file exists
    if not os.path.isfile(file_path):
        print("File not found:", file_path)
        return []

    # Load stories from the file
    stories = []
    with open(file_path, 'r', encoding='utf-8') as file:
        for line in file:
            # Process each line (assuming each line contains a story)
            story, highlights = split_story(line)
            stories.append({'story': story, 'highlights': highlights})
    return stories

# Load stories
file_path = '/kaggle/input/newspaper-text-summarization-cnn-dailymail/cnn_dailymail/train.csv'
stories = load_stories(file_path)
print('Loaded Highlight Stories:', len(stories))

Loaded Highlight Stories: 1088106


* The number 1,088,106 represents the count of stories that have been successfully loaded and processed